src/include/stir/recon_buildblock/CUDA/CudaGibbsPenalty.h

```diff
@@ -0,0 +1,139 @@
+/*
+    Copyright (C) 2025, University College London
+    Copyright (C) 2025, University of Milano-Bicocca
+
+    SPDX-License-Identifier: Apache-2.0
+
+    See STIR/LICENSE.txt for details
+*/
+/*!
+  \file
+  \ingroup priors
+  \ingroup CUDA
+  \brief Declaration of the stir::CudaGibbsPenalty class
+
+  \author Kris Thielemans
+  \author Matteo Neel Colombo
+*/
+
+#ifndef __stir_recon_buildblock_CUDA_CudaGibbsPenalty_H__
+#define __stir_recon_buildblock_CUDA_CudaGibbsPenalty_H__
+
+#include "stir/Array.h"
+#include "stir/DiscretisedDensity.h"
+
+#ifndef __CUDACC__
+struct dim3
+{
+  unsigned int x = 1, y = 1, z = 1;
+};
+struct int3
+{
+  int x = 0, y = 0, z = 0;
+};
+#else
+#  include <cuda_runtime.h>
+#endif
+
+#include "stir/shared_ptr.h"
+#include <string>
```

Comment:
What do you think about moving this to `cuda_utilities.h`, and including that instead? Might be useful for others later. I think it will prevent conflicts if we need to define `dim3` etc elsewhere as well, but as you prefer.